### ISSUE TITLE
Failed to prepare request to OAuth server

### ISSUE THREAD

**Issue by iromli** [GluuFederation/oxIdp#2]
After upgrading to oxIDP v2.4.3, the oxIDP container (docker edition) got error as seen in this gist: https://gist.github.com/iromli/8f368cbc497778f2f8d361f2af9c4273.

This issue happened when an integration script in SP (i.e. https://sp.example.com/protected/printHeaders.py) tried to access https://ox.example.com/idp/Authn/RemoteUser in IDP. Note that all config have been synchronized with CE 2.4.3.

Was there any missing config?

**Comment by mzico** [comment]
@iromli : What error do you get in /opt/idp/logs/idp-process.log ?

**Comment by yurem** [comment]
@iromli Can you try to use [this version](http://ox.gluu.org/maven/org/xdi/oxidp/2.4.3-SNAPSHOT/oxidp-2.4.3-SNAPSHOT.war) ? It's based in Shib 2.4.5. Also logs will help us.

**Comment by iromli** [comment]
@yurem I have rebuilt the image using 2.4.3.Final.war (build date 2016-04-03). It seems the error still happened.

Here's the contents of `idp-process.log`:

```
07:25:36.900 - TRACE [edu.internet2.middleware.shibboleth.idp.session.IdPSessionFilter:117] - Attempting to retrieve IdP session cookie.
07:25:36.910 - INFO [Shibboleth-Access:73] - 20160404T072536Z|10.2.0.5|ox.example.com:443|/profile/SAML2/Redirect/SSO|
07:25:36.913 - DEBUG [edu.internet2.middleware.shibboleth.idp.profile.IdPProfileHandlerManager:86] - shibboleth.HandlerManager: Looking up profile hand
ler for request path: /SAML2/Redirect/SSO
07:25:36.913 - DEBUG [edu.internet2.middleware.shibboleth.idp.profile.IdPProfileHandlerManager:97] - shibboleth.HandlerManager: Located profile handler
 of the following type for the request path: edu.internet2.middleware.shibboleth.idp.profile.saml2.SSOProfileHandler
07:25:36.914 - DEBUG [edu.internet2.middleware.shibboleth.idp.util.HttpServletHelper:339] - LoginContext key cookie was not present in request
07:25:36.914 - DEBUG [edu.internet2.middleware.shibboleth.idp.profile.saml2.SSOProfileHandler:188] - Incoming request does not contain a login context,
 processing as first leg of request
07:25:36.915 - DEBUG [edu.internet2.middleware.shibboleth.idp.profile.saml2.SSOProfileHandler:366] - Decoding message with decoder binding 'urn:oasis:n
ames:tc:SAML:2.0:bindings:HTTP-Redirect'
07:25:36.943 - DEBUG [edu.internet2.middleware.shibboleth.common.relyingparty.provider.SAMLMDRelyingPartyConfigurationManager:128] - Looking up relying
 party configuration for https://sp.example.com/shibboleth
07:25:36.944 - DEBUG [edu.internet2.middleware.shibboleth.common.relyingparty.provider.SAMLMDRelyingPartyConfigurationManager:134] - No custom relying 
party configuration found for https://sp.example.com/shibboleth, looking up configuration based on metadata groups.
07:25:36.944 - DEBUG [edu.internet2.middleware.shibboleth.common.relyingparty.provider.SAMLMDRelyingPartyConfigurationManager:157] - No custom or group
-based relying party configuration found for https://sp.example.com/shibboleth. Using default relying party configuration.
07:25:36.944 - DEBUG [edu.internet2.middleware.shibboleth.common.relyingparty.RelyingPartySecurityPolicyResolver:117] - Resolving security policy based
 on communication profile ID: urn:mace:shibboleth:2.0:profiles:saml2:sso
07:25:36.947 - DEBUG [edu.internet2.middleware.shibboleth.idp.profile.saml2.SSOProfileHandler:387] - Decoded request from relying party 'https://sp.exa
mple.com/shibboleth'
07:25:36.947 - DEBUG [edu.internet2.middleware.shibboleth.common.relyingparty.provider.SAMLMDRelyingPartyConfigurationManager:128] - Looking u[30/1834]
 party configuration for https://sp.example.com/shibboleth
07:25:36.947 - DEBUG [edu.internet2.middleware.shibboleth.common.relyingparty.provider.SAMLMDRelyingPartyConfigurationManager:134] - No custom relying 
party configuration found for https://sp.example.com/shibboleth, looking up configuration based on metadata groups.
07:25:36.947 - DEBUG [edu.internet2.middleware.shibboleth.common.relyingparty.provider.SAMLMDRelyingPartyConfigurationManager:157] - No custom or group
-based relying party configuration found for https://sp.example.com/shibboleth. Using default relying party configuration.
07:25:36.947 - DEBUG [edu.internet2.middleware.shibboleth.idp.profile.saml2.SSOProfileHandler:226] - Creating login context and transferring control to
 authentication engine
07:25:36.951 - DEBUG [edu.internet2.middleware.shibboleth.idp.util.HttpServletHelper:181] - Storing LoginContext to StorageService partition loginConte
xts, key 3a5a372459cf056abc2fd7e9665aefe4fdc0348c525102c1fa6458fdce9320d2
07:25:36.952 - DEBUG [edu.internet2.middleware.shibboleth.idp.profile.saml2.SSOProfileHandler:240] - Redirecting user to authentication engine at https
://ox.example.com:443/idp/AuthnEngine
07:25:36.996 - TRACE [edu.internet2.middleware.shibboleth.idp.session.IdPSessionFilter:117] - Attempting to retrieve IdP session cookie.
07:25:36.996 - DEBUG [edu.internet2.middleware.shibboleth.idp.authn.AuthenticationEngine:209] - Processing incoming request
07:25:36.996 - TRACE [edu.internet2.middleware.shibboleth.idp.util.HttpServletHelper:349] - Looking up LoginContext with key 3a5a372459cf056abc2fd7e966
5aefe4fdc0348c525102c1fa6458fdce9320d2 from StorageService parition: loginContexts
07:25:36.996 - TRACE [edu.internet2.middleware.shibboleth.idp.util.HttpServletHelper:355] - Retrieved LoginContext with key 3a5a372459cf056abc2fd7e9665
aefe4fdc0348c525102c1fa6458fdce9320d2 from StorageService parition: loginContexts
07:25:36.997 - DEBUG [edu.internet2.middleware.shibboleth.idp.authn.AuthenticationEngine:240] - Beginning user authentication process.
07:25:36.997 - DEBUG [edu.internet2.middleware.shibboleth.idp.authn.AuthenticationEngine:283] - Filtering configured LoginHandlers: {urn:oasis:names:tc
:SAML:2.0:ac:classes:PreviousSession=edu.internet2.middleware.shibboleth.idp.authn.provider.PreviousSessionLoginHandler@14e6134d, urn:oasis:names:tc:SA
ML:2.0:ac:classes:unspecified=edu.internet2.middleware.shibboleth.idp.authn.provider.RemoteUserLoginHandler@cfd2ef3, urn:oasis:names:tc:SAML:2.0:ac:cla
sses:PasswordProtectedTransport=edu.internet2.middleware.shibboleth.idp.authn.provider.UsernamePasswordLoginHandler@42155a2e}
07:25:36.997 - DEBUG [edu.internet2.middleware.shibboleth.idp.authn.AuthenticationEngine:332] - Filtering out previous session login handler because th
ere is no existing IdP session
07:25:36.997 - DEBUG [edu.internet2.middleware.shibboleth.idp.authn.AuthenticationEngine:464] - Selecting appropriate login handler from filtered set {
urn:oasis:names:tc:SAML:2.0:ac:classes:unspecified=edu.internet2.middleware.shibboleth.idp.authn.provider.RemoteUserLoginHandler@cfd2ef3, urn:oasis:nam
es:tc:SAML:2.0:ac:classes:PasswordProtectedTransport=edu.internet2.middleware.shibboleth.idp.authn.provider.UsernamePasswordLoginHandler@42155a2e}
07:25:36.998 - DEBUG [edu.internet2.middleware.shibboleth.idp.authn.AuthenticationEngine:497] - Authenticating user with login handler of type edu.inte
rnet2.middleware.shibboleth.idp.authn.provider.RemoteUserLoginHandler
07:25:36.998 - DEBUG [edu.internet2.middleware.shibboleth.idp.authn.provider.RemoteUserLoginHandler:66] - Redirecting to https://ox.example.com:443/idp
/Authn/RemoteUser
07:25:37.040 - TRACE [edu.internet2.middleware.shibboleth.idp.session.IdPSessionFilter:117] - Attempting to retrieve IdP session cookie.
07:25:37.457 - ERROR [org.gluu.oxauth.client.authentication.AuthenticationFilter:80] - Failed to preapre request to OAuth server
java.lang.NullPointerException: null
        at org.gluu.oxauth.client.util.Configuration.getPropertyValue(Configuration.java:85) ~[saml-openid-auth-client-2.4.3.Final.jar:na]
        at org.gluu.oxauth.client.util.AbstractConfigurationFilter.getPropertyFromInitParams(AbstractConfigurationFilter.java:50) ~[saml-openid-auth-cl
ient-2.4.3.Final.jar:na]
        at org.gluu.oxauth.client.authentication.AuthenticationFilter.getOAuthRedirectUrl(AuthenticationFilter.java:119) ~[saml-openid-auth-client-2.4.
3.Final.jar:na]
        at org.gluu.oxauth.client.authentication.AuthenticationFilter.doFilter(AuthenticationFilter.java:78) ~[saml-openid-auth-client-2.4.3.Final.jar:
na]
        at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:241) [catalina.jar:7.0.65]
        at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:208) [catalina.jar:7.0.65]
        at edu.internet2.middleware.shibboleth.idp.session.IdPSessionFilter.doFilter(IdPSessionFilter.java:87) [shibboleth-identityprovider-2.4.5.jar:n
a]
        at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:241) [catalina.jar:7.0.65]
        at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:208) [catalina.jar:7.0.65]
        at edu.internet2.middleware.shibboleth.common.log.SLF4JMDCCleanupFilter.doFilter(SLF4JMDCCleanupFilter.java:52) [shibboleth-common-1.4.5.jar:na
]
        at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:241) [catalina.jar:7.0.65]
        at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:208) [catalina.jar:7.0.65]
        at org.apache.catalina.core.StandardWrapperValve.invoke(StandardWrapperValve.java:203) [catalina.jar:7.0.65]
        at org.apache.catalina.core.StandardContextValve.invoke(StandardContextValve.java:122) [catalina.jar:7.0.65]
        at org.apache.catalina.authenticator.AuthenticatorBase.invoke(AuthenticatorBase.java:505) [catalina.jar:7.0.65]
        at org.apache.catalina.core.StandardHostValve.invoke(StandardHostValve.java:170) [catalina.jar:7.0.65]
        at org.apache.catalina.valves.ErrorReportValve.invoke(ErrorReportValve.java:103) [catalina.jar:7.0.65]
        at org.apache.catalina.core.StandardEngineValve.invoke(StandardEngineValve.java:116) [catalina.jar:7.0.65]
        at org.apache.catalina.connector.CoyoteAdapter.service(CoyoteAdapter.java:423) [catalina.jar:7.0.65]
        at org.apache.coyote.ajp.AjpProcessor.process(AjpProcessor.java:190) [tomcat-coyote.jar:7.0.65]
        at org.apache.coyote.AbstractProtocol$AbstractConnectionHandler.process(AbstractProtocol.java:625) [tomcat-coyote.jar:7.0.65]
        at org.apache.tomcat.util.net.JIoEndpoint$SocketProcessor.run(JIoEndpoint.java:316) [tomcat-coyote.jar:7.0.65]
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1145) [na:1.7.0_95]
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:615) [na:1.7.0_95]
        at org.apache.tomcat.util.threads.TaskThread$WrappingRunnable.run(TaskThread.java:61) [tomcat-coyote.jar:7.0.65]
        at java.lang.Thread.run(Thread.java:745) [na:1.7.0_95]
```

And here's the `oxidp-ldap.properties`:

```
bindDN: cn=directory manager
bindPassword: ppCLB9cV1IU=
servers: ldap.gluu.local:1636
useSSL: true
maxconnections: 3

oxidp_ConfigurationEntryDN=ou=oxidp,ou=configuration,inum=@!0854.5657.36AD.3A95!0002!0932.E1A1,ou=appliances,o=gluu
confDir=

binaryAttributes=objectGUID
```

Given those config and latest war, `/idp/Authn/RemoteUser` URL gave me blank white page.

**Comment by iromli** [comment]
After setting up `DEBUG` log level, I saw more log messages in `/opt/idp/logs/idp-process.log` as seen below:

```
16:23:57.762 - TRACE [edu.internet2.middleware.shibboleth.idp.session.IdPSessionFilter:117] - Attempting to retrieve IdP session cookie.     [102/1848]
16:23:57.768 - INFO [Shibboleth-Access:73] - 20160404T162357Z|10.2.0.5|ox.example.com:443|/profile/SAML2/Redirect/SSO|
16:23:57.770 - DEBUG [edu.internet2.middleware.shibboleth.idp.profile.IdPProfileHandlerManager:86] - shibboleth.HandlerManager: Looking up profile hand
ler for request path: /SAML2/Redirect/SSO
16:23:57.770 - DEBUG [edu.internet2.middleware.shibboleth.idp.profile.IdPProfileHandlerManager:97] - shibboleth.HandlerManager: Located profile handler
 of the following type for the request path: edu.internet2.middleware.shibboleth.idp.profile.saml2.SSOProfileHandler
16:23:57.771 - DEBUG [edu.internet2.middleware.shibboleth.idp.util.HttpServletHelper:339] - LoginContext key cookie was not present in request
16:23:57.771 - DEBUG [edu.internet2.middleware.shibboleth.idp.profile.saml2.SSOProfileHandler:188] - Incoming request does not contain a login context,
 processing as first leg of request
16:23:57.771 - DEBUG [edu.internet2.middleware.shibboleth.idp.profile.saml2.SSOProfileHandler:366] - Decoding message with decoder binding 'urn:oasis:n
ames:tc:SAML:2.0:bindings:HTTP-Redirect'
16:23:57.797 - DEBUG [PROTOCOL_MESSAGE:113] - 
<?xml version="1.0" encoding="UTF-8"?>
<samlp:AuthnRequest
    AssertionConsumerServiceURL="https://sp.example.com/Shibboleth.sso/SAML2/POST"
    Destination="https://ox.example.com/idp/profile/SAML2/Redirect/SSO"
    ID="_c2630b5319bb7bb988fecb0ed33945e9"
    IssueInstant="2016-04-04T16:25:29Z"
    ProtocolBinding="urn:oasis:names:tc:SAML:2.0:bindings:HTTP-POST"
    Version="2.0" xmlns:samlp="urn:oasis:names:tc:SAML:2.0:protocol">
    <saml:Issuer xmlns:saml="urn:oasis:names:tc:SAML:2.0:assertion">https://sp.example.com/shibboleth</saml:Issuer>
    <samlp:NameIDPolicy AllowCreate="1"/>
</samlp:AuthnRequest>
16:23:57.798 - DEBUG [edu.internet2.middleware.shibboleth.common.relyingparty.provider.SAMLMDRelyingPartyConfigurationManager:128] - Looking u[78/1848]
 party configuration for https://sp.example.com/shibboleth
16:23:57.798 - DEBUG [edu.internet2.middleware.shibboleth.common.relyingparty.provider.SAMLMDRelyingPartyConfigurationManager:134] - No custom relying 
party configuration found for https://sp.example.com/shibboleth, looking up configuration based on metadata groups.
16:23:57.798 - DEBUG [edu.internet2.middleware.shibboleth.common.relyingparty.provider.SAMLMDRelyingPartyConfigurationManager:157] - No custom or group
-based relying party configuration found for https://sp.example.com/shibboleth. Using default relying party configuration.
16:23:57.798 - DEBUG [edu.internet2.middleware.shibboleth.common.relyingparty.RelyingPartySecurityPolicyResolver:117] - Resolving security policy based
 on communication profile ID: urn:mace:shibboleth:2.0:profiles:saml2:sso
16:23:57.813 - DEBUG [edu.internet2.middleware.shibboleth.idp.profile.saml2.SSOProfileHandler:387] - Decoded request from relying party 'https://sp.exa
mple.com/shibboleth'
16:23:57.813 - DEBUG [edu.internet2.middleware.shibboleth.common.relyingparty.provider.SAMLMDRelyingPartyConfigurationManager:128] - Looking up relying
 party configuration for https://sp.example.com/shibboleth
16:23:57.813 - DEBUG [edu.internet2.middleware.shibboleth.common.relyingparty.provider.SAMLMDRelyingPartyConfigurationManager:134] - No custom relying 
party configuration found for https://sp.example.com/shibboleth, looking up configuration based on metadata groups.
16:23:57.814 - DEBUG [edu.internet2.middleware.shibboleth.common.relyingparty.provider.SAMLMDRelyingPartyConfigurationManager:157] - No custom or group
-based relying party configuration found for https://sp.example.com/shibboleth. Using default relying party configuration.
16:23:57.814 - DEBUG [edu.internet2.middleware.shibboleth.idp.profile.saml2.SSOProfileHandler:226] - Creating login context and transferring control to
 authentication engine
16:23:57.818 - DEBUG [edu.internet2.middleware.shibboleth.idp.util.HttpServletHelper:181] - Storing LoginContext to StorageService partition loginConte
xts, key bf441f9b463aff40ea0313a6cd4a6ba5abe29820948cf1a4f643e049dc2c183d
16:23:57.819 - DEBUG [edu.internet2.middleware.shibboleth.idp.profile.saml2.SSOProfileHandler:240] - Redirecting user to authentication engine at https
://ox.example.com:443/idp/AuthnEngine
16:23:57.867 - TRACE [edu.internet2.middleware.shibboleth.idp.session.IdPSessionFilter:117] - Attempting to retrieve IdP session cookie.
16:23:57.868 - DEBUG [edu.internet2.middleware.shibboleth.idp.authn.AuthenticationEngine:209] - Processing incoming request
16:23:57.868 - TRACE [edu.internet2.middleware.shibboleth.idp.util.HttpServletHelper:349] - Looking up LoginContext with key bf441f9b463aff40ea0313a6cd
4a6ba5abe29820948cf1a4f643e049dc2c183d from StorageService parition: loginContexts
16:23:57.868 - TRACE [edu.internet2.middleware.shibboleth.idp.util.HttpServletHelper:355] - Retrieved LoginContext with key bf441f9b463aff40ea0313a6cd4
a6ba5abe29820948cf1a4f643e049dc2c183d from StorageService parition: loginContexts
16:23:57.868 - DEBUG [edu.internet2.middleware.shibboleth.idp.authn.AuthenticationEngine:240] - Beginning user authentication process.
16:23:57.868 - DEBUG [edu.internet2.middleware.shibboleth.idp.authn.AuthenticationEngine:283] - Filtering configured LoginHandlers: {urn:oasis:names:tc
:SAML:2.0:ac:classes:PreviousSession=edu.internet2.middleware.shibboleth.idp.authn.provider.PreviousSessionLoginHandler@3e9e08b9, urn:oasis:names:tc:SA
ML:2.0:ac:classes:unspecified=edu.internet2.middleware.shibboleth.idp.authn.provider.RemoteUserLoginHandler@794ec9cc, urn:oasis:names:tc:SAML:2.0:ac:cl
asses:PasswordProtectedTransport=edu.internet2.middleware.shibboleth.idp.authn.provider.UsernamePasswordLoginHandler@23a2ae4c}
16:23:57.869 - DEBUG [edu.internet2.middleware.shibboleth.idp.authn.AuthenticationEngine:332] - Filtering out previous session login handler because th
ere is no existing IdP session
16:23:57.869 - DEBUG [edu.internet2.middleware.shibboleth.idp.authn.AuthenticationEngine:464] - Selecting appropriate login handler from filte[43/1848]
urn:oasis:names:tc:SAML:2.0:ac:classes:unspecified=edu.internet2.middleware.shibboleth.idp.authn.provider.RemoteUserLoginHandler@794ec9cc, urn:oasis:na
mes:tc:SAML:2.0:ac:classes:PasswordProtectedTransport=edu.internet2.middleware.shibboleth.idp.authn.provider.UsernamePasswordLoginHandler@23a2ae4c}
16:23:57.869 - DEBUG [edu.internet2.middleware.shibboleth.idp.authn.AuthenticationEngine:497] - Authenticating user with login handler of type edu.inte
rnet2.middleware.shibboleth.idp.authn.provider.RemoteUserLoginHandler
16:23:57.869 - DEBUG [edu.internet2.middleware.shibboleth.idp.authn.provider.RemoteUserLoginHandler:66] - Redirecting to https://ox.example.com:443/idp
/Authn/RemoteUser
16:23:57.911 - TRACE [edu.internet2.middleware.shibboleth.idp.session.IdPSessionFilter:117] - Attempting to retrieve IdP session cookie.
16:23:57.914 - DEBUG [org.gluu.oxauth.client.authentication.AuthenticationFilter:71] - No code and no OAuth data found
16:23:57.949 - DEBUG [org.apache.commons.configuration.PropertiesConfiguration:588] - FileName set to /opt/tomcat/conf/oxidp-ldap.properties
16:23:57.951 - DEBUG [org.apache.commons.configuration.ConfigurationUtils:449] - ConfigurationUtils.locate(): base is null, name is /opt/tomcat/conf/ox
idp-ldap.properties
16:23:57.952 - DEBUG [org.apache.commons.configuration.DefaultFileSystem:281] - Could not locate file /opt/tomcat/conf/oxidp-ldap.properties at null: n
o protocol: /opt/tomcat/conf/oxidp-ldap.properties
16:23:57.952 - DEBUG [org.apache.commons.configuration.ConfigurationUtils:471] - Loading configuration from the absolute path /opt/tomcat/conf/oxidp-ld
ap.properties
16:23:57.953 - DEBUG [org.apache.commons.configuration.PropertiesConfiguration:626] - Base path set to file:///opt/tomcat/conf/oxidp-ldap.properties
16:23:57.973 - DEBUG [org.apache.commons.configuration.PropertiesConfiguration:588] - FileName set to /opt/tomcat/conf/salt
16:23:57.974 - DEBUG [org.apache.commons.configuration.ConfigurationUtils:449] - ConfigurationUtils.locate(): base is null, name is /opt/tomcat/conf/sa
lt
16:23:57.974 - DEBUG [org.apache.commons.configuration.DefaultFileSystem:281] - Could not locate file /opt/tomcat/conf/salt at null: no protocol: /opt/
tomcat/conf/salt
16:23:57.974 - DEBUG [org.apache.commons.configuration.ConfigurationUtils:471] - Loading configuration from the absolute path /opt/tomcat/conf/salt
16:23:57.974 - DEBUG [org.apache.commons.configuration.PropertiesConfiguration:626] - Base path set to file:///opt/tomcat/conf/salt
16:23:58.113 - DEBUG [org.gluu.site.ldap.LDAPConnectionProvider:127] - Using next binary attributes: [objectguid]
16:23:58.370 - DEBUG [org.gluu.oxauth.client.conf.Configuration:199] - Created LdapEntryManager: org.gluu.site.ldap.persistence.LdapEntryManager@320413
ad
16:23:58.370 - INFO [org.gluu.oxauth.client.conf.Configuration:165] - Loading configuration from Ldap...
16:23:58.396 - DEBUG [org.gluu.site.ldap.persistence.AbstractEntryManager:1004] - LdapProperty: application, AttributeName: oxConfApplication,[15/1848]
eValue: [{
    "applicationName":"Saml",

    "openIdProviderUrl":"https://ox.example.com/.well-known/openid-configuration",
    "openIdClientId":"@!0854.5657.36AD.3A95!0001!B765.78AE!0008!A524.CF00",
    "openIdClientPassword":"MY02xNCI9BcPEUfSBOCxkg==",
    "openIdScopes":[
        "openid",
        "profile",
        "email",
        "user_name"
    ],
    "openIdRedirectUrl": "https://localhost:8443/identity/authentication/finishlogout"
}]
16:23:58.475 - DEBUG [org.gluu.site.ldap.persistence.AbstractEntryManager:1004] - LdapProperty: revision, AttributeName: oxRevision, AttributeValue: [1
]
16:23:58.475 - INFO [org.gluu.oxauth.client.conf.Configuration:90] - Configuration loaded successfully.
16:23:58.487 - ERROR [org.gluu.oxauth.client.authentication.AuthenticationFilter:80] - Failed to preapre request to OAuth server
java.lang.NullPointerException: null
        at org.gluu.oxauth.client.util.Configuration.getPropertyValue(Configuration.java:85) ~[saml-openid-auth-client-2.4.3.Final.jar:na]
        at org.gluu.oxauth.client.util.AbstractConfigurationFilter.getPropertyFromInitParams(AbstractConfigurationFilter.java:50) ~[saml-openid-auth-cl
ient-2.4.3.Final.jar:na]
        at org.gluu.oxauth.client.authentication.AuthenticationFilter.getOAuthRedirectUrl(AuthenticationFilter.java:119) ~[saml-openid-auth-client-2.4.
3.Final.jar:na]
        at org.gluu.oxauth.client.authentication.AuthenticationFilter.doFilter(AuthenticationFilter.java:78) ~[saml-openid-auth-client-2.4.3.Final.jar:
na]
        at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:241) [catalina.jar:7.0.65]
        at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:208) [catalina.jar:7.0.65]
        at edu.internet2.middleware.shibboleth.idp.session.IdPSessionFilter.doFilter(IdPSessionFilter.java:87) [shibboleth-identityprovider-2.4.5.jar:n
a]
        at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:241) [catalina.jar:7.0.65]
        at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:208) [catalina.jar:7.0.65]
        at edu.internet2.middleware.shibboleth.common.log.SLF4JMDCCleanupFilter.doFilter(SLF4JMDCCleanupFilter.java:52) [shibboleth-common-1.4.5.jar:na
]
        at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:241) [catalina.jar:7.0.65]
        at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:208) [catalina.jar:7.0.65]
        at org.apache.catalina.core.StandardWrapperValve.invoke(StandardWrapperValve.java:203) [catalina.jar:7.0.65]
        at org.apache.catalina.core.StandardContextValve.invoke(StandardContextValve.java:122) [catalina.jar:7.0.65]
        at org.apache.catalina.authenticator.AuthenticatorBase.invoke(AuthenticatorBase.java:505) [catalina.jar:7.0.65]
        at org.apache.catalina.core.StandardHostValve.invoke(StandardHostValve.java:170) [catalina.jar:7.0.65]
        at org.apache.catalina.valves.ErrorReportValve.invoke(ErrorReportValve.java:103) [catalina.jar:7.0.65]
        at org.apache.catalina.core.StandardEngineValve.invoke(StandardEngineValve.java:116) [catalina.jar:7.0.65]
        at org.apache.catalina.connector.CoyoteAdapter.service(CoyoteAdapter.java:423) [catalina.jar:7.0.65]
        at org.apache.coyote.ajp.AjpProcessor.process(AjpProcessor.java:190) [tomcat-coyote.jar:7.0.65]
        at org.apache.coyote.AbstractProtocol$AbstractConnectionHandler.process(AbstractProtocol.java:625) [tomcat-coyote.jar:7.0.65]
        at org.apache.tomcat.util.net.JIoEndpoint$SocketProcessor.run(JIoEndpoint.java:316) [tomcat-coyote.jar:7.0.65]
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1145) [na:1.7.0_95]
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:615) [na:1.7.0_95]
        at org.apache.tomcat.util.threads.TaskThread$WrappingRunnable.run(TaskThread.java:61) [tomcat-coyote.jar:7.0.65]
        at java.lang.Thread.run(Thread.java:745) [na:1.7.0_95]
```

Any hints on where I should look into?

**Comment by iromli** [comment]
Recently I found that with oxIdp v2.4.2 (https://ox.gluu.org/maven/org/xdi/oxidp/2.4.2.Final/) and latest config taken from CE v2.4.3, I didn't encounter this issue.

**Comment by zamilskhan** [comment]
Faced the same error in 2.4.3 beta-3 testing aginst Shib SP.

> 20:29:55.838 - INFO [Shibboleth-Access:73] - 20160414T202955Z|192.168.122.1|centos.gluu.info:443|/profile/SAML2/Redirect/SSO|
20:29:55.929 - ERROR [org.gluu.oxauth.client.authentication.AuthenticationFilter:80] - Failed to preapre request to OAuth server
java.lang.NullPointerException: null
	at org.gluu.oxauth.client.util.Configuration.getPropertyValue(Configuration.java:85) ~[saml-openid-auth-client-2.4.3.Final.jar:na]
	at org.gluu.oxauth.client.util.AbstractConfigurationFilter.getPropertyFromInitParams(AbstractConfigurationFilter.java:50) ~[saml-openid-auth-client-2.4.3.Final.jar:na]

**Comment by yurem** [comment]
Fixed in CE 2.4.3 beta4

**Comment by iromli** [comment]
Thanks, it's fixed.
